### DISCUSSION TITLE
Most efficient way to install zsh-vim-mode?

### DISCUSSION THREAD

**Discussion by lkhphuc** [marlonrichert/zsh-snap#46] [Q&A] [answered]
Current I only use these prompt config:
```
znap prompt sindresorhus/pure
znap prompt softmoth/zsh-vim-mode
```
The prompt show up pretty much instantly but the blinking cursor from zsh-vim-mode took a bit longer to show up. I notice that in `~znap` there is a file called `prompt_zsh-vim-mode_setup` created every time I open a new shell.
The content of that file is just `.znap.source zsh-vim-mode`.

I think there is probably some inefficiency here and wonder if there is any better way to handle this.

If I install with `znap source zsh-vim-mode` instead of `.. prompt` the shell crashed and quited.

Another small issue is that functionalities of `pure` shell such as git status and VI-mode indicator doesn't work  
![Screenshot 2021-02-19 at 02 31 45](https://user-images.githubusercontent.com/12573521/108448981-9d808200-725a-11eb-80c3-b6a728f15571.png)
One work around is to add another line so that it becomes:
```
znap prompt sindresorhus/pure
znap prompt softmoth/zsh-vim-mode
znap prompt sindresorhus/pure
```
But then again this will loose the blinking bar cursor from the vim-mode.

**Comment by lkhphuc** [comment, marked as answer]
To avoid conflict of two `znap prompt` command, I now switched to `powerlevel10k` according to #31 and use p10k 'instant prompt' feature instead.
Now zshrc doesn't contains any `znap prompt` statement:
```
# p10k instant prompt code ....
...

znap source romkatv/powerlevel10k
znap source softmoth/zsh-vim-mode
```
Everything works smoothly for this combination, instant prompt and blinking bar cursor.

**Reply by marlonrichert** [reply]
I am not really sure what you're talking about, but good that you solved your problem. 🙂
